### ISSUE TITLE
Parent decoder name used in ossec-logtest instead the child decoder name

### ISSUE THREAD

**Issue by jmv74211** [wazuh/wazuh#4937]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.1-1 | ossec-logtest | Manager | Packages | Centos 7 |

Hello team,

Working on the response to a community user, I have found that when creating child decoders and using the `ossec-logtest` tool to debug them, the name of the parent decoder appears, and not the 
child decoder that is the one that is actually being used.

Here is an example:

![imagen](https://user-images.githubusercontent.com/23462183/79771380-ffb4e180-832e-11ea-8265-f17ade7485ee.png)

This can cause confusion since the user has also told me the case (See https://groups.google.com/d/msg/wazuh/oaf2mUOwW0g/tjf_5ieIAQAJ)

I think the most appropriate thing would be to show the name of the child decoder, which is the one that is actually being used, as it happens with the rules

Best regards.

**Comment by jnasselle** [comment]
Hi!

This was solved in [wazuh-logtest](https://github.com/wazuh/wazuh/pull/6103). Closing this issue.

Regards,
Nico

**Comment by ptrsimon** [comment]
Hi @jnasselle,

Are you sure this is resolved? I'm pretty sure that the child decoder is the one matching:
```
<decoder name="sshd">
  <program_name>^sshd</program_name>
</decoder>

<decoder name="ssh-closed-2">
  <parent>sshd</parent>
  <prematch>^Connection closed by </prematch>
  <regex offset="after_prematch">^(\S+) port (\S+)</regex>
  <order>srcip, srcport</order>
</decoder>
```

The output from wazuh-logtest:
```
Starting wazuh-logtest v4.1.5
Type one log per line

Jul 29 09:07:58 testhost sshd[1111]: Connection closed by 1.1.1.1 port 1111 [preauth]

**Phase 1: Completed pre-decoding.
	full event: 'Jul 29 09:07:58 testhost sshd[1111]: Connection closed by 1.1.1.1 port 1111 [preauth]'
	timestamp: 'Jul 29 09:07:58'
	hostname: 'testhost'
	program_name: 'sshd'

**Phase 2: Completed decoding.
	name: 'sshd'
	parent: 'sshd'
	srcip: '1.1.1.1'
	srcport: '1111'

**Phase 3: Completed filtering (rules).
	id: '5722'
	level: '0'
	description: 'sshd: ssh connection closed.'
	groups: '['syslog', 'sshd']'
	firedtimes: '1'
	gdpr: '['IV_32.2']'
	hipaa: '['164.312.b']'
	mail: 'False'
	nist_800_53: '['AU.14', 'AC.7']'
	pci_dss: '['10.2.5']'
	tsc: '['CC6.8', 'CC7.2', 'CC7.3']'
```

I think in this case the decoder name showed by phase 2 should be ssh-closed-2.

I'm using Wazuh 4.1.5, maybe this fix was introduced in a newer version?

**Comment by jnasselle** [comment]
Hi @ptrsimon ,

Sorry about not being more descriptive in my closing comment. In ossec-logtest there was not impact of [`use_own_name`](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/decoders.html#use-own-name) decoder option when evaluating rules, while `wazuh-logtest` show it appropriately BUT you still have to set that option in you decoder.
```xml
<decoder name="sshd">
  <program_name>^sshd</program_name>
</decoder>

<decoder name="ssh-closed-2">
  <parent>sshd</parent>
  <prematch>^Connection closed by </prematch>
  <regex offset="after_prematch">^(\S+) port (\S+)</regex>
  <order>srcip, srcport</order>
  <use_own_name>true</use_own_name>
</decoder>
```
- `ossec-logtest`
```
 /var/ossec/bin/ossec-logtest 
2021/07/29 10:13:04 ossec-testrule: INFO: Started (pid: 11561).

ossec-testrule: Type one log per line.

Jul 29 09:07:58 testhost sshd[1111]: Connection closed by 1.1.1.1 port 1111 [preauth]


**Phase 1: Completed pre-decoding.
       full event: 'Jul 29 09:07:58 testhost sshd[1111]: Connection closed by 1.1.1.1 port 1111 [preauth]'
       timestamp: 'Jul 29 09:07:58'
       hostname: 'testhost'
       program_name: 'sshd'
       log: 'Connection closed by 1.1.1.1 port 1111 [preauth]'

**Phase 2: Completed decoding.
       decoder: 'sshd'
       srcip: '1.1.1.1'
       srcport: '1111'

```
- `wazuh-logtest`
```
/var/ossec/bin/wazuh-logtest 
Starting wazuh-logtest v4.1.5
Type one log per line

Jul 29 09:07:58 testhost sshd[1111]: Connection closed by 1.1.1.1 port 1111 [preauth]

**Phase 1: Completed pre-decoding.
        full event: 'Jul 29 09:07:58 testhost sshd[1111]: Connection closed by 1.1.1.1 port 1111 [preauth]'
        timestamp: 'Jul 29 09:07:58'
        hostname: 'testhost'
        program_name: 'sshd'

**Phase 2: Completed decoding.
        name: 'ssh-closed-2'
        parent: 'sshd'
        srcip: '1.1.1.1'
        srcport: '1111'
```
Now it shows `ssh-closed-2` as decoder name, making to not match rule 5700 (095-sshd_rules.xml) `decoded_as` match criteria since the decoder name now is not `sshd`
 
Regards,
Nico

**Comment by ptrsimon** [comment]
Hi @jnasselle,

Thank you for the detailed explanation, my bad for not noticing this option in the docs. This also clarifies why `use_own_name` isn't enabled by default.

Best regards,
Peter
